CONTRIBUTING.md

```diff
@@ -1,45 +1,111 @@
 
-## Feature requests and ideas
+To edit the documentation, edit the `docs/*.md` files on the **develop** branch. You can see the result if you run `make html` (or `poetry run mkdocs serve` if on Windows) inside the project's root directory, then navigating your browser to [locahost:8000](http://localhost:8000).
 
-So, you have an idea for a great feature? Awesome! We'd love to hear from you! Please [open a new issue](https://github.com/jrnl-org/jrnl/issues/new/choose) and describe the goal of the feature, and any relevant use cases. We'll discuss the issue with you, and decide if it's a good fit for the project.
+The `gh-pages` branch is automatically maintained and generated after your changes are merged. You should never have to edit that branch.
+
+### Recipes and external editors
+
+If you'd like to share a jrnl command line trick that you find useful, or advice on how to integrate a particular external editor, you may find it worthwile to add it to the ["Recipes" section](docs/recipes.md).
+
+## Testing
+
+Much of the work of maintaining jrnl involves testing rather than coding.
+
+The nature of jrnl means we deal with extremely sensitive data, and can't risk data loss. While jrnl does have a comprehensive automated testing suite, user testing is crucial to mitigating this risk.
+
+### Prereleases
+
+[Prereleases are deployed through PyPi much like normal releases](https://pypi.org/project/jrnl/#history). You can use [pipx](https://pypi.org/project/pipx/) to fetch them and test them. See the [changelog](CHANGELOG.md) for information on what has changed with each release.
+
+### Pull requests
+
+If you are comfortable enough with git, feel free to fetch particular [pull requests](https://github.com/jrnl-org/jrnl/pulls), test them yourself, and report back your findings. Bonus points if you can add a screencast of how the new feature works.
+
+### Confirm bug reports
+
+There are always [open bugs among our GitHub issues](https://github.com/jrnl-org/jrnl/issues?q=is%3Aissue+is%3Aopen+label%3Abug) and many are specific to a particular OS, Python version, or jrnl version. A simple comment like "Confirmed on jrnl v2.2, MacOS 10.15, Python 3.8.1" would be extremely helpful in tracking down bugs.
+
+### Automate tests
+
+See the develop section below for information on how to contribute new automated tests.
+
+## Submitting feature requests and ideas
+
+If you have a feature request or idea for jrnl, please [open a new issue](https://github.com/jrnl-org/jrnl/issues/new/choose) and describe the goal of the feature, and any relevant use cases. We'll discuss the issue with you, and decide if it's a good fit for the project.
 
 When discussing new features, please keep in mind our design goals. jrnl strives to do one thing well. To us, that means:
 
 * be _slim_
 * have a simple interface
 * avoid duplicating functionality
 
-## New programmers and programmers new to python
+## Developing
```

Comment:
Can we add something to this section about looking in the backlog, and what the various labels are for?

Reply:
Added a "Finding things to work on" section